src/main/kotlin/demo/controllers/OrganizationController.kt

```diff
@@ -0,0 +1,49 @@
+package demo.controllers
+
+import demo.models.people.Organization
+import demo.services.OrganizationService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import java.util.*
+
+@RestController
+@RequestMapping("/api/organizations")
+class OrganizationController(@Autowired private val organizationService: OrganizationService) {
+    @GetMapping
+    fun getOrganizations(): ResponseEntity<MutableIterable<Organization>> {
+        return ResponseEntity(organizationService.getOrganizations(), HttpStatus.OK)
+    }
+
+    @GetMapping("/{organizationId}")
+    fun getOrganization(@PathVariable organizationId: String): ResponseEntity<Organization> {
+        val organization: Optional<Organization> = organizationService.getOrganization(organizationId)
+        return if (organization.isPresent)
+            ResponseEntity(organizationService.getOrganization(organizationId).get(), HttpStatus.OK)
+        else
+            ResponseEntity(null, HttpStatus.NOT_FOUND)
+    }
+
+    @PostMapping("/addorganization")
+    fun addOrganization(@RequestBody organization: Organization): ResponseEntity<Organization> {
```

Comment:
Co to za klasa Organization? Jeśli jest to bezpośrednio model, to nie powinno tu tak być. Tak jak gadaliśmy na dsie, powinieneś mieć jakąś klasę OrganizationRequest / OrganizationDto i to mapować na obiekty domenowe.

Reply:
Nawet jeśli atrybutami są typy prymitywne i dodanie klasy pośredniej nie daje nam nic?

Reply:
Na razie może nie, ale still - jeśli chcemy potem iść w Domain Driven Development, to jak najbardziej. Tam są potem takie bajki, że nie używasz typów prymitywnych - ogólnie zamiast `organizationName: String` robisz wtedy osobną data classę, którą nazywasz `OrganizationName` z jednym tylko polem. Daje to tyle, że dokładnie wiesz, co jest czym i ciężej się jebnąć (nawet jeśli obecne IDE mają dobre podpowiedzi, to i tak możesz z nich naumyślnie nie korzystać). To się jakoś nazywa, ale nie pamiętam teraz nazwy.